src/Orleans.Runtime/MembershipService/MembershipOracle.cs

```diff
@@ -12,7 +12,7 @@ namespace Orleans.Runtime.MembershipService
 {
     internal class MembershipOracle : SystemTarget, IMembershipOracle, IMembershipService
     {
-        private readonly static TimeSpan shutdownGossipTimeout = TimeSpan.FromMilliseconds(30);
+        private readonly static TimeSpan shutdownGossipTimeout = TimeSpan.FromSeconds(10);
```

Comment:
This seems too drastic of a change.

Reply:
I disagree, it's just that we need time to send one request per silo (in parallel). 30ms was way too short, so we always had a timeout, even though the task was not cancelled.

Reply:
The timeout is benign noise, right? The gossip call goes in parallel to all silos telling them to reread the table. While 30ms is clearly too aggressive for that, wouldn't 500ms or 1s be more than enough of time for silos to read the table?

Reply:
Up to 5s sounds reasonable to me